### PR TITLE
Multiple wizard skills bugfixes

### PR DESCRIPTION
**Pull request by nitoygo**
- Fixed teleport bug after teleporting ally
  - Removed usage of 'Teleport' global variable
- Fixed nova skill not activating

### CODE REVIEW

File: Source Main 5.2/source/WSclient.cpp [hunk 1]
```diff
@@ -218,8 +218,6 @@ int MoveCount = 0;
 
 int CurrentSkill = 0;
 
-bool Teleport = false;
```
Comment (nitoygo) [inline Source Main 5.2/source/WSclient.cpp:221]: This is the root of all evil. Better to remove it. In the server side, this is tied to the player, while here, it is used for everything that is teleporting.

File: Source Main 5.2/source/ZzzInterface.cpp [hunk 2]
```diff
@@ -6149,16 +6148,8 @@ void AttackWizard(CHARACTER* c, int Skill, float Distance)
...
-                    if (Teleport)
-                    {
-                        Teleport = false;
-                    }
-                    else
-                    {
-                        Teleport = true;
-                        SocketClient->ToGameServer()->SendTeleportTarget(tc->Key, TargetX, TargetY);
-                        SetPlayerTeleport(tc);
-                    }
```
Comment (nitoygo) [inline Source Main 5.2/source/ZzzInterface.cpp:6161]: Removing the Teleport boolean checks here is the actual fix to the teleport bug, but I think it's better to remove the Teleport variable altogether. if ever it's needed, it's best to check o->Teleport value instead.

File: Source Main 5.2/source/NewUIMainFrameWindow.cpp [hunk 3]
```diff
@@ -1832,6 +1832,12 @@ int SEASON3B::CNewUISkillList::GetHotKey(int iHotKey)
 
 int SEASON3B::CNewUISkillList::GetSkillIndex(int iSkillType)
 {
+    // special handling for skills with different skill id for the trigger
+    if (iSkillType == AT_SKILL_BLAST_HELL_BEGIN)
+    {
+        iSkillType = AT_SKILL_BLAST_HELL;
+    }
```
Comment (nitoygo) [inline Source Main 5.2/source/NewUIMainFrameWindow.cpp:1839]: Taking inspiration from the comment here:
https://github.com/sven-n/MuMain/issues/78#issue-2725294683
but with a slightly different approach

File: Source Main 5.2/source/ZzzInterface.cpp (hunk 2 above)
Comment (sven-n) [reply Source Main 5.2/source/ZzzInterface.cpp:6161]: Yes, global variables are a bad idea anyway 👍

**Review by sven-n — APPROVED** [review]

**Pull request merged**